spec/query_spec.cr

```diff
@@ -746,14 +746,10 @@ describe Avram::Query do
 
       users = UserQuery.new.group(&.age).group(&.id)
       users.query.statement.should eq "SELECT #{User::COLUMN_SQL} FROM users GROUP BY users.age, users.id"
-      users.map(&.name).should contain "Dwight"
+      users.map(&.name).should eq ["Dwight", "Michael", "Jim"]
```

Comment:
Oh so does it just return an array? What does group by do in this case. I thought it'd be a hash or something

Reply:
It returns an array here of `.map()`, but `UserQuery.new.group(&.age).group(&.id)` just returns the `BaseQuery` object like normal. It is confusing though because technically in this query, the group is useless. 

I see this more as building blocks for the next parts to come as well as the underlying access. But if you have suggestions, or whatever, I'm all ears. Postgres grouping hurts my head lol